documentation/src/main/resources/_posts/2023-08-23-ditto-benchmark.md

```diff
@@ -0,0 +1,465 @@
+---
+title: "Eclipse Ditto Benchmark"
+published: false
+permalink: 2023-08-23-ditto-benchmark.html
+layout: post
+author: vvasilevbosch
+tags: [blog]
+hide_sidebar: true
+sidebar: false
+toc: true
+
+---
+
+This blog post is presenting a benchmark of Eclipse Ditto. It consists of a few scenarios to cover most important ditto functionalities, test the performance and provide a tuning guide. This benchmark is done via the [benchmark-tool](https://github.com/bosch-io/ditto/tree/master/benchmark-tool), based on [k6](https://k6.io/) load testing tool
+
+## Setup and used tools
+
+- EKS cluster using [m5.4xlarge](https://aws.amazon.com/ec2/instance-types/m5/) nodes.
+
+- Deployed Ditto using [ditto's official helm chart](https://artifacthub.io/packages/helm/eclipse-iot/ditto).
+
+- Deployed [bitnami]([Helm Charts to deploy Apache Kafka in Kubernetes](https://bitnami.com/stack/kafka/helm)) kafka helm chart.
+
+- Deployed [MMock](https://github.com/jmartin82/mmock) instance - used as a 'dummy' receiver for ditto events and device commands.
+
+- [k6](https://k6.io/) - load testing tool, used to implement the benchmark scenarios.
+
+- Deployed [k6-operator]([GitHub - grafana/k6-operator: An operator for running distributed k6 tests.](https://github.com/grafana/k6-operator)) - Kubernetes operator for running distributed k6 tests.
+
+- [MongoDB](https://cloud.mongodb.com/) instance of type [M50](https://www.mongodb.com/docs/atlas/manage-clusters/#nvme-considerations)
+
+## Scenarios
+
+The benchmark test consists of 4 test scenarios that can be executed independently and in parallel:
+
+- [READ_THINGS](#read-things) - read things via HTTP ( get things by id )
+
+- [SEARCH_THINGS](#search-things) - search things via HTTP ( get things by applying search filter )
+
+- [MODIFY_THINGS](#modify-things) - Modify things by sending ditto protocol kafka messages to specfic topic. Ditto kafka connection is reading from this topic and processes the messages. Ditto HTTP push connection is configured in ditto, which sends events from topic **/things/twin/events?filter=eq(topic:action,'modified')** to a monster mock endpoint, which replies with HTTP status code 204.
+
+- [DEVICE_LIVE_MESSAGES](#device-live-messagescommands) - Send live messages to things via HTTP. Ditto HTTP push connection is configured, which sends events from topic **/things/live/messages** to a monster mock endpoint, which replies with predefined ditto protocol message.
+
+## Preparation
+
+1 000 000 things are created with the [benchmark-tool](https://github.boschdevcloud.com/bosch-iot-things/ditto/tree/feature/ditto-benchmark-test) by sending Ditto protocol messages to the topic that a Ditto Kafka connection is configured to read from. This connection has an [implicitThingCreation mapper](https://eclipse.dev/ditto/connectivity-mapping.html#implicitthingcreation-mapper) configured.
```

Comment:
The link to the benchmark tool points to a Bosch internal Github.

Reply:
Instead of linking to `[implicitThingCreation mapper](https://eclipse.dev/ditto/connectivity-mapping.html#implicitthingcreation-mapper)` it would be better to use relative links - so that this also still works if/when the EclipseFoundation decides to change domains again ..

So instead: [implicitThingCreation mapper](connectivity-mapping.html#implicitthingcreation-mapper)